src/Api/Request.php

```diff
@@ -32,17 +34,17 @@ public function setConfig(array $configuration) {
    * @throws \Exception
    */
   public function getAuth() {
-    // Verify necessary configuration is available.
-    if (empty($this->configuration['api_information']['username'])
-    || empty($this->configuration['api_information']['password'])
-    || empty($this->configuration['api_information']['customer_number'])) {
+    $config = $this->configuration['api_information'];
+
+    if (empty($config['username']) || empty($config['password']) || empty($config['customer_number']) || empty($this->configuration['api_information']['mode'])) {
       throw new \Exception('Configuration is required.');
     }
 
     return [
       'username' => $this->configuration['api_information']['username'],
       'password' => $this->configuration['api_information']['password'],
       'customer_number' => $this->configuration['api_information']['customer_number'],
+      'mode' => $this->configuration['api_information']['mode'],
     ];
   }
 
```

Comment:
No need for a base class as each service (TrackingService, RateService etc.) will provide its own methods and the authentication is not to be stored in properties.

We may simply need an interface for each service.